### PR TITLE
Added dt_pivot_wider() and dt_pivot_longer()

### PR DESCRIPTION
**Pull request by markfairbanks**
Here's my attempt at these! Thanks for walking me through the process

### CODE REVIEW

File: R/pivot_longer.R [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+#' @param names_to Name of the new "names" column. Must be a string.
+#' @param values_to Name of the new "values" column. Must be a string.
+#'
+#' @export
+#'
+#' @examples
+#' example_df <- data.table(x = c(1,2,3), y = c(4,5,6), z = c("a", "b", "c"))
+#'
+#' example_df %>%
+#'   dt_pivot_longer(cols = c(x, y), names_to = "stuff", values_to = "things")
```
Comment (TysonStanley) [inline R/pivot_longer.R]: Add `library(data.table)` at the beginning of the examples.

File: R/pivot_longer.R [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+                            names_to = "name",
+                            values_to = "value") {
+
+  if (missing(cols)) {
+    # All columns if cols = NULL
+    cols <- colnames(data)
+  } else {
+    cols <- characterize(substitute(cols))
+  }
+
```
Comment (TysonStanley) [inline R/pivot_longer.R]: Near the top of the function, consider adding something like:

```
if (isFALSE(is.data.table(data)))
  data <- as.data.table(data)
```

File: R/pivot_longer.R [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  id_vars <- colnames(data)[!colnames(data) %in% cols]
+
+  data.table::melt(data = data,
+                   id.vars = id_vars,
+                   measure.vars = cols,
+                   variable.name = names_to,
+                   value.name = values_to,
+                   # ...,
```
Comment (TysonStanley) [inline R/pivot_longer.R]: I like the idea of using `...` to be able to pass additional arguments to `melt()`.

File: R/pivot_wider.R [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+#'
+#' @export
+#'
+#' @examples
+#' example_dt <- data.table(z = rep(c("a", "b", "c"), 2),
+#'                          stuff = c(rep("x", 3), rep("y", 3)),
+#'                          things = 1:6)
+#'
+#' example_dt %>%
+#'   dt_pivot_wider(names_from = stuff, values_from = things)
```
Comment (TysonStanley) [inline R/pivot_wider.R]: Add `library(data.table)` here too :)

File: R/pivot_wider.R [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+dt_pivot_wider <- function(data,
+                           id_cols = NULL,
+                           names_from = name,
+                           names_sep = "_",
+                           values_from = value) {
+
```
Comment (TysonStanley) [inline R/pivot_wider.R]: I think this function is particularly helpful as I don't think `dcast()` is intuitive. Love what you've done here.

File: R/pivot_wider.R [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+characterize <- function(vec_list_expr) {
+  vle_length <- length(vec_list_expr)
+  if (vle_length == 1) {
+    as.character(vec_list_expr)
+  } else if (as.character(vec_list_expr)[1] == "-"){
+    as.character(vec_list_expr)
+  } else {
+    as.character(vec_list_expr)[-1]
+  }
+}
```
Comment (TysonStanley) [inline R/pivot_wider.R]: This doesn't need to be repeated in both files. In fact, you could put this function in the `utils.R` file.

**Review by TysonStanley — CHANGES_REQUESTED** [review]
This is a fantastic start, @mtfairbanks! Thanks for putting this together. I've left some feedback if you are interested in making these adjustments. 

Another thing to consider is creating unit tests. You can see how I've done that thus far in `tidyfast` by looking in the `tests` folder. These can help make sure the functions are doing what we expect them to do in a variety of cases.

**Comment by markfairbanks** [comment]
Thanks for the suggestions! I'll try and work on some unit tests this weekend. Happy thanksgiving

**Comment by markfairbanks** [comment]
Any thoughts on the latest commits?

**Comment by TysonStanley** [comment]
I like how these look, @mtfairbanks! I can totally work with this. I very much appreciate your additions here. I'll let you know when they are fully integrated into the package. 

Note that a list of the contributors (like yourself) will be highlighted in the README. Thanks again!

**Pull request merged**